### DISCUSSION TITLE
Which package to install with my firebase app

### DISCUSSION THREAD

**Discussion by ajubin** [invertase/react-native-firebase#4454] [Q&A] [answered]
I've got a fixed version of `@react-native-firebase/app` alongside with `crashlytics`.

I then decided to add `perf` and `dynamic-links` but I cannot guess which version to install based on app version. 

My solution was to look for in history of `tests` to find appropriate version. https://github.com/invertase/react-native-firebase/commits/master/tests/package.json. Install the package, then when installing the pod on iOS, check that I have no warnings.

Is there any solution to know it more easily ?

**Comment by mikehardy** [comment, marked as answer]
Install and use the most recent version of all of them.
And in general stay up to date with the most recent version of all of them.
Don't overthink it past that.

**Reply by ajubin** [reply]
thanks for the reply. 

That's indeed a good idea 👍  When I installed crashlytics, they were some bugs with reporting on iOS, so I installed the latest stable version. 
Currently in my projet, upgrading all libraries is not automatic and has a cost. So we upgrade them from time to time. 

How do you deal with this in your projects ?

**Reply by mikehardy** [reply]
I eat the cost as needed. I typically release weekly and I typically upgrade everything to latest immediately post release / git tag followed by a quick round of testing and backing down things that didn't work. This keeps me close to current as a normal state so the rate of change ingested isn't so high ongoing. I'm pretty handy with patch-package and not afraid to PR dependencies though, this may not work in a slower-pace-of-change org. In this case, for react-native-firebase, it pays to be current I think. perf app links and crashlytics either haven't moved much recently so shouldn't be a big deal (perf app links) or have a bugfix for test crashes (crashlytics) that you'll want current for

**Reply by ajubin** [reply]
I was struggling to see the changelog when upgrading multiple versions, and I finally found a solution.

open changelog files in every packages and check latest issues with these packages.

Thanks for your answers mikehardy
